src/utils/internal.ts

```diff
@@ -1174,3 +1197,237 @@ export function assertTickerValid(ticker: string): void {
+    const [rawClaim, issuer] = transferCondition.asClaimOwnership;
+    const restrictionIssuerDid = identityIdToString(issuer);
+    const claim = statsClaimToStatClaimUserType(rawClaim);
+    if (restrictionIssuerDid === issuerDid && claim.type === claimType) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/**
+ * @hidden
+ */
+function getClaimType(statClaim: PolymeshPrimitivesStatisticsStatClaim): ClaimType {
+  if (statClaim.isAccredited) {
+    return ClaimType.Accredited;
+  } else if (statClaim.isAffiliate) {
+    return ClaimType.Affiliate;
+  } else {
+    return ClaimType.Jurisdiction;
+  }
+}
+
+/**
+ * @hidden
+ */
+function compareOptionalBigNumbers(a: BigNumber | undefined, b: BigNumber | undefined): boolean {
+  if (a === undefined && b === undefined) {
+    return true;
+  }
+  if (a === undefined || b === undefined) {
+    return false;
+  }
+  return a.eq(b);
+}
+
+/**
+ * @hidden
+ */
+export function compareTransferRestrictionToInput(
+  transferRestriction: PolymeshPrimitivesTransferComplianceTransferCondition,
+  value: BigNumber | ClaimCountRestrictionValue,
+  type: TransferRestrictionType
+): boolean {
+  if (transferRestriction.isMaxInvestorCount && type === TransferRestrictionType.Count) {
+    const currentCount = u64ToBigNumber(transferRestriction.asMaxInvestorCount);
+    return currentCount.eq(value as BigNumber);
+  } else if (
+    transferRestriction.isMaxInvestorOwnership &&
+    type === TransferRestrictionType.Percentage
+  ) {
+    const currentOwnership = permillToBigNumber(transferRestriction.asMaxInvestorOwnership);
+    return currentOwnership.eq(value as BigNumber);
+  } else if (transferRestriction.isClaimCount && type === TransferRestrictionType.ClaimCount) {
+    const [statClaim, rawIssuerId, rawMin, maybeMax] = transferRestriction.asClaimCount;
+    const issuerDid = identityIdToString(rawIssuerId);
+    const min = u64ToBigNumber(rawMin);
+    const max = maybeMax.isSome ? u64ToBigNumber(maybeMax.unwrap()) : undefined;
+    const castedValue = value as ClaimCountRestrictionValue;
+
+    return !!(
+      castedValue.min.eq(min) &&
+      compareOptionalBigNumbers(max, castedValue.max) &&
+      castedValue.claim.type === getClaimType(statClaim) &&
+      issuerDid === castedValue.issuer.did
+    );
+  } else if (
+    transferRestriction.isClaimOwnership &&
+    type === TransferRestrictionType.ClaimPercentage
+  ) {
+    const castedValue = value as ClaimCountRestrictionValue;
+    const [statClaim, rawIssuerId, rawMin, rawMax] = transferRestriction.asClaimOwnership;
+    const issuerDid = identityIdToString(rawIssuerId);
+
+    const min = permillToBigNumber(rawMin);
+    const max = permillToBigNumber(rawMax);
+
+    return !!(
+      castedValue.min.eq(min) &&
+      compareOptionalBigNumbers(castedValue.max, max) &&
+      castedValue.claim.type === getClaimType(statClaim) &&
+      issuerDid === castedValue.issuer.did
+    );
+  }
```

Comment:
There's a lot of repeated code between the last two blocks, you should be able to reuse some of it, or at least destructure `castedValue` to avoid repeating it so much